tools/serverpod_cli/lib/src/generator/dart/library_generators/server_test_tools_generator.dart

```diff
@@ -181,6 +181,33 @@ class ServerTestToolsGenerator {
       'package:${config.serverPackage}/src/generated/protocol.dart',
     );
 
+    Spec handleParameter(ParameterDefinition parameterDef) {
+      if (parameterDef.type.isRecordType) {
+        return mapRecordToJsonRef.call([refer(parameterDef.name)]).code;
+      }
+
+      if (parameterDef.type.returnsRecordInContainer) {
+        return Block.of([
+          if (parameterDef.type.nullable)
+            Code('${parameterDef.name} == null ? null :'),
+          mapRecordContainingContainerToJsonRef
+              .call([refer(parameterDef.name)]).code,
+        ]);
+      }
+
+      if (parameterDef.type.isMapType &&
+          parameterDef.type.generics.first.className != 'String') {
+        return Block.of([
+          Code([
+            if (parameterDef.type.nullable) '${parameterDef.name} != null && ',
+            '${parameterDef.name}.isEmpty ? [] : ${parameterDef.name}'
+          ].join())
```

Comment:
Comment: Same thing here does this handle nested Maps? I feel like some recursion might be in order.

Reply:
Nesting was indeed a problem, which is now fixed by generalizing the "map containers with records" to also include a mapping for containers/records which at some point contain a non-String keyed map.